docs/conf.py

```
@@ -396,15 +397,66 @@ def render(self, template_name, context):
+    fetches inventories with its own user-agent and may succeed where a bare
+    probe is blocked, so only connection-level failures (DNS, refused, timeout)
+    count as "offline". This avoids wrongly disabling intersphinx in CI just
+    because a CDN rejects the probe's user-agent.
+    """
+    import urllib.error  # noqa: PLC0415
+    import urllib.request  # noqa: PLC0415
+
+    for base, _ in mapping.values():
+        url = base.rstrip("/") + "/objects.inv"
+        try:
+            with urllib.request.urlopen(url, timeout=timeout):
```

Ruff (0.15.15)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)